src/libraries/System.Linq.AsyncEnumerable/src/System/Linq/Empty.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Linq
+{
+    public static partial class AsyncEnumerable
+    {
+        /// <summary>
+        /// Returns an empty <see cref="IAsyncEnumerable{T}"/> that has the specified type argument.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the elements of the sequence.</typeparam>
+        /// <returns>An empty <see cref="IAsyncEnumerable{T}"/> whose type argument is <typeparamref name="TResult"/>.</returns>
+        public static IAsyncEnumerable<TResult> Empty<TResult>() => EmptyAsyncEnumerable<TResult>.Instance;
```

Comment:
It's not the responsibility of this library, but would it be reasonable to support collection literals for IAE?

Reply:
For testing purposes, it'd be reasonable. I don't know how much value it would add beyond that. Maybe as a shorthand for Empty, as you allude to here with your comment placement.

Reply:
I think testing is a strong enough reason to consider this. The currently available solutions are simply too verbose.

Reply:
My preference then would be that IAsyncEnumerable is special-cased in the compiler, just as is IEnumerable. Worth a conversation with @CyrusNajmabadi and co. But if we collectively decide not to do it in the compiler, we'd add a builder method somewhere.